GW2EIEvtcParser/EncounterLogic/Strikes/SotO/TempleOfFebe.cs

```diff
@@ -128,29 +140,76 @@ internal override void EIEvtcParse(ulong gw2Build, FightData fightData, AgentDat
             int curRegret = 1;
             foreach (AbstractSingleActor target in Targets)
             {
-                if (target.IsSpecies(ArcDPSEnums.TrashID.EmbodimentOfDespair))
+                switch (target.ID)
                 {
-                    target.OverrideName(target.Character + " " + curDespair++);
-                }
-                if (target.IsSpecies(ArcDPSEnums.TrashID.EmbodimentOfEnvy))
-                {
-                    target.OverrideName(target.Character + " " + curEnvy++);
-                }
-                if (target.IsSpecies(ArcDPSEnums.TrashID.EmbodimentOfGluttony))
-                {
-                    target.OverrideName(target.Character + " " + curGluttony++);
-                }
-                if (target.IsSpecies(ArcDPSEnums.TrashID.EmbodimentOfMalice))
-                {
-                    target.OverrideName(target.Character + " " + curMalice++);
-                }
-                if (target.IsSpecies(ArcDPSEnums.TrashID.EmbodimentOfRage))
-                {
-                    target.OverrideName(target.Character + " " + curRage++);
-                }
-                if (target.IsSpecies(ArcDPSEnums.TrashID.EmbodimentOfRegret))
-                {
-                    target.OverrideName(target.Character + " " + curRegret++);
+                    case (int)ArcDPSEnums.TrashID.EmbodimentOfDespair:
+                        CombatItem despair = combatData.FirstOrDefault(x => x.SkillID == EmpoweredDespairEmbodiment && x.DstAgent == target.AgentItem.Agent);
```

Comment:
agent matching should use DstMatchesAgent API on CombatItem.cs (same for the ones below), also use IsBuffApply API on CombatItem.cs to be safe.